middleware/cluster.go

```diff
@@ -0,0 +1,45 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/dskit/clusterutil"
+)
+
+// ClusterValidationMiddleware validates that requests have the correct cluster verification label.
+func ClusterValidationMiddleware(cluster string, auxPaths []string, invalidClusters *prometheus.CounterVec, logger log.Logger) Interface {
+	var reB strings.Builder
+	// Allow for a potential path prefix being configured.
+	reB.WriteString(".*/(metrics|debug/pprof.*|ready")
```

Comment:
I think this MUST be configurable. Each downstream will have some paths that may not need the cluster verification, for example the admin frontend for Mimir.

Reply:
Maybe that's what `auxPaths` is? I think this method needs more docs.

Reply:
`auxPaths` is for additional paths, yes. I kept a base set of hard-coded paths here since they are common (`/metrics`, `/debug/pprof`, `/ready`).